### PR TITLE
Hide zip menu item when there is no create permission in a folder

### LINKED ISSUE #39: Error while Zipping when root folder is read only: `No create permission for path`
Just tried the app and I got the following issue:
```
{"reqId":"ddnrK6weTbpLI0qMPo7o","level":3,"time":"2021-11-19T00:55:02+01:00","remoteAddr":"","user":"--","app":"files_zip","method":"","url":"--","message":"Failed to create zip archive","userAgent":"--","version":"22.2.3.0","exception":{"Exception":"OCP\\Files\\NotPermittedException","Message":"No create permission for path","Code":0,"Trace":[{"file":"/var/www/nextcloud/apps/files_zip/lib/Service/ZipService.php","line":92,"function":"newFile","class":"OC\\Files\\Node\\Folder","type":"->"},{"file":"/var/www/nextcloud/apps/files_zip/lib/BackgroundJob/ZipJob.php","line":65,"function":"zip","class":"OCA\\FilesZip\\Service\\ZipService","type":"->"},{"file":"/var/www/nextcloud/lib/public/BackgroundJob/Job.php","line":79,"function":"run","class":"OCA\\FilesZip\\BackgroundJob\\ZipJob","type":"->"},{"file":"/var/www/nextcloud/lib/public/BackgroundJob/QueuedJob.php","line":47,"function":"execute","class":"OCP\\BackgroundJob\\Job","type":"->"},{"file":"/var/www/nextcloud/cron.php","line":127,"function":"execute","class":"OCP\\BackgroundJob\\QueuedJob","type":"->"}],"File":"/var/www/nextcloud/lib/private/Files/Node/Folder.php","Line":209,"CustomMessage":"Failed to create zip archive"},"id":"6196e81a535c6"}
```
It has most likely the same reason why https://github.com/nextcloud/server/issues/25787 happened.

### PR DESCRIPTION
**Pull request by juliushaertl**

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -29,6 +29,15 @@ import { showError, showSuccess } from '@nextcloud/dialogs'
 					).then(this.enhancePrompt.bind(this, suggestedFilename))
 				},
 			})
+
+			fileList.$el.on('urlChanged', data => {
```
Comment (PVince81) [inline src/main.js:33]: you might be able to combine both permissions in a single string

**Review by PVince81 — APPROVED** [review]
:+1: see minor comment

**Review by szaimen — COMMENTED** [review]
@juliushaertl thanks for the fix! The thing is that I have actually create permissions in the folder I was trying to create the zip archive in. But the root folder itself is read only on my instance which probably triggered the issue although it should work in the directory I am in... 
(see https://github.com/nextcloud/server/issues/25787 where something like this happened to)

**Comment by juliushaertl** [comment]
> @juliushaertl thanks for the fix! The thing is that I have actually create permissions in the folder I was trying to create the zip archive in. But the root folder itself is read only on my instance which probably triggered the issue although it should work in the directory I am in...
> (see nextcloud/server#25787 where something like this happened to)

Oh then let me double check the original issue, the PR makes sense nevertheless, I'll address the comment from Vincent

**Comment by juliushaertl** [comment]
Lets get this in and follow up on the original ticket

**Pull request merged**